include/revng/PipelineC/Prototypes.h

```diff
@@ -401,6 +403,27 @@ rp_analysis_get_argument_name(rp_analysis *analysis, int index);
 int rp_analysis_get_argument_acceptable_kinds_count(rp_analysis *analysis,
                                                     int argument_index);
 
+/**
+ * \return the ammout of extra arguments of the provided analysis
+ */
+int rp_analysis_get_options_count(rp_analysis *analysis);
+
+/**
+ * \return the name of the extra argument with the provided index. Returns
+ * nullptr if extra_argument_index < 0 or extra_argument_index >
+ * rp_analysis_get_extra_argument_count(analysis)
+ */
+const char * /*owning*/
+rp_analysis_get_option_name(rp_analysis *analysis, int extra_argument_index);
+
+/**
+ * \return the type of the extra argument with the provided index. Returns
+ * nullptr if extra_argument_index < 0 or extra_argument_index >
+ * rp_analysis_get_extra_argument_count(analysis)
+ */
+const char * /*owning*/
+rp_analysis_get_option_type(rp_analysis *analysis, int extra_argument_index);
```

Comment:
You sure about `owning`? Can't we return a reference to a static string?

Reply:
the issue is that it is implemented as getting a vector of strings, each rappresenting the type or name of the argument. To make it return a non-owing pointer would require two virtual fuctions in the type erased analysis, one to get the number of options and the other to retrieve it. It is not impossible, just uglier.